File: src/main/scala/br/com/fabioaraujo/calcula/produto/service/ProdutoService.scala
package br.com.fabioaraujo.calcula.produto.service

import br.com.fabioaraujo.calcula.produto.dao.ProdutoRepository
import br.com.fabioaraujo.calcula.produto.dto.Produto
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ProdutoService(@Autowired val produtoRepository: ProdutoRepository) {

  def listar(): java.lang.Iterable[Produto] = {
    produtoRepository.findAll
  }

  def getProduto(id: Long): Produto = {
    produtoRepository.findOne(id)
  }

  def criar(produto: Produto): Long = {
    produtoRepository.save(produto)
    produto.id
  }
}

File: src/main/java/br/com/fabioaraujo/calcula/produto/service/ProdutoService.java
package br.com.fabioaraujo.calcula.produto.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.fabioaraujo.calcula.produto.dao.ProdutoRepository;
import br.com.fabioaraujo.calcula.produto.dto.Produto;

@Service
public class ProdutoService {
	@Autowired 
	private ProdutoRepository produtoRepository;
	 

  public Iterable<Produto> listar(){
    return produtoRepository.findAll();
  }

  public Produto getProduto(Long id) {
    return produtoRepository.findById(id).orElse(null);
  }

  public Long criar(Produto produto) {
    produto = produtoRepository.save(produto);
    return produto.getId();
  }
}